### PR TITLE
Do not export CategoricalPool

### PR DESCRIPTION
**Pull request by bkamins**
following Issue #108.

### CODE REVIEW

**Comment by bkamins** [comment]
@nalimilan build errors seem unrelated

File: test/01_typedef.jl [hunk 1]
```diff
@@ -1,7 +1,8 @@
 module TestTypeDef
 using Base.Test
 using CategoricalArrays
-using CategoricalArrays: DefaultRefType, level,  reftype, leveltype, catvalue, iscatvalue
+using CategoricalArrays: DefaultRefType, level, reftype, leveltype, catvalue, iscatvalue
+using CategoricalArrays: CategoricalPool
```
Comment (nalimilan) [inline test/01_typedef.jl]: Put this on the previous line.

Comment (bkamins) [reply test/01_typedef.jl]: fixed, although it produces overlong line

Comment (nalimilan) [reply test/01_typedef.jl]: Yeah, but who said we are not allowed to wrap lines? ;-)

**Comment by alyst** [comment]
@nalimilan Yes, fine with me. Thanks for consideration.

It may make sense to use `CategoricalPool` as a definition of a specific category (levels + ordered flag + potential read-only levels flag).
But atm there's not enough support for it on API side, e.g. "==" or pool sharing across arrays. I guess there's not much demand for it atm either.

File: test/01_typedef.jl (hunk 1 above)
Comment (bkamins) [reply test/01_typedef.jl]: 👍

**Comment by nalimilan** [comment]
Ah, too bad, it appears that `CategoricalPool` is already used by several dependencies without importing it explicitly: http://juliarun-ci.s3.amazonaws.com/dc6c2265352f2424efe36dd39c5f3edf454eccce/CategoricalArrays_reverse_dependency_tests.html

I've reverted the commit and re-tagged the release without this change. Given the amount of painful breakage we have inflicted on dependencies recently, I don't think it's worth introducing yet another breaking change for this. Looks like we'll have to treat `CategoricalPool` as part of the public API (at least until the next major release).

**Comment by alyst** [comment]
It looks like most of the breakage is still within JuliaData.
I can fix it for RData anticipating that the withdrawal of CategoricalPool will happen in the next minor release of CategoricalArrays.

**Comment by nalimilan** [comment]
I guess we should decide whether we really want to unexport `CategoricalPool` in the next major release, or whether we want to define a public API for it. Maybe our time would be more productively spent doing the latter, given that packages do use `CategoricalPool` alread (though probably to a limited extent).

**Pull request merged**